src/AppBundle/Controller/CategoryController.php

```diff
@@ -31,8 +31,26 @@ public function categoryDetail(Request $request)
 			throw new NotFoundHttpException("Kategorie neexistuje");
 		}
 
+		// paginator
+        $em = $this->get('doctrine.orm.entity_manager');
+        $dql = 'SELECT p
```

Comment:
Uvažuješ správně, určitě pokud je jeden kód duplikovaný, tak má cenu ho dát dohromady. Můžeš si tedy vytvořit metodu, která ten dotaz staví a ve findByCategory vrátit getResult nad jejím výsledkem